### PR TITLE
Firmware fixes and improvements for flashing via blackmagic

### PR DESCRIPTION
**Pull request by quen0n**
# What's new

- Improved detection of blackmagic probe by USB
- Skipping the requirement to press "c" when flashing through blackmagic

# Verification 

- Flash the Flipper with blackmagic

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: scripts/fbt_tools/blackmagic.py [hunk 1]
```diff
@@ -27,8 +27,12 @@ def _find_probe(self):
             raise StopError("More than one Blackmagic probe found")
         else:
             # If you're getting any issues with auto lookup, uncomment this
-            # print("\n".join([f"{p.device} {vars(p)}" for p in ports]))
-            return sorted(ports, key=lambda p: f"{p.location}_{p.name}")[0]
+            #print("\n".join([f"{p.device} {vars(p)}" for p in ports]))
+            port_list = sorted(ports, key=lambda p: f"{p.location}_{p.name}")
+            for port in port_list:
+                if str(port.location) == "None":
```
Comment (hedger) [inline scripts/fbt_tools/blackmagic.py]: Can you please provide explanation and examples for this case?

Comment (quen0n) [reply scripts/fbt_tools/blackmagic.py]: In this case, port.location - is a sign of the Blackmagic debug port. The port for logs has a specific value. Thus, it is possible to detect the Blackmagic port, regardless of the order in which the ports were installed in the system.

```
COM2 {'device': 'COM2', 'name': 'COM2', 'description': 'Устройство с последовательным интерфейсом USB (COM2)', 'hwid': 'USB VID:PID=303A:4001 SER=BLACKMAGIC', 'vid': 12346, 'pid': 16385, 'serial_number': 'BLACKMAGIC', 'location': None, 'manufacturer': 'Корпорация Майкрософт', 'product': None, 'interface': None}
COM3 {'device': 'COM3', 'name': 'COM3', 'description': 'Устройство с последовательным интерфейсом USB (COM3)', 'hwid': 'USB VID:PID=303A:4001 SER=BLACKMAGIC LOCATION=1-3:x.2', 'vid': 12346, 'pid': 16385, 'serial_number': 'BLACKMAGIC', 'location': '1-3:x.2', 'manufacturer': 'Корпорация Майкрософт', 'product': None, 'interface': None}      
```
Here are two com ports that are determined when connecting blackmagic. Here you can compare locations.

**Review by hedger — DISMISSED** [review]

File: scripts/fbt_tools/blackmagic.py (hunk 1 above)
Comment (skotopes) [reply scripts/fbt_tools/blackmagic.py]: Why not to use `if port.location is None:` ? str(None) looks a little bit too much...

**Comment by skotopes** [comment]
also un-draft when ready

File: scripts/fbt_tools/blackmagic.py (hunk 1 above)
Comment (hedger) [reply scripts/fbt_tools/blackmagic.py]: On my system, all ports have a valid Locaiton:
```
./fbt get_blackmagic
COM15 {'device': 'COM15', 'name': 'COM15', 'description': 'Устройство с последовательным интерфейсом USB (COM15)', 'hwid': 'USB VID:PID=303A:4001 SER=BLACKMAGIC LOCATION=1-5:x.2', 'vid': 12346, 'pid': 16385, 'serial_number': 'BLACKMAGIC', 'location': '1-5:x.2', 'manufacturer': 'Microsoft', 'product': None, 'interface': None}
COM16 {'device': 'COM16', 'name': 'COM16', 'description': 'Устройство с последовательным интерфейсом USB (COM16)', 'hwid': 'USB VID:PID=303A:4001 SER=BLACKMAGIC LOCATION=1-5:x.0', 'vid': 12346, 'pid': 16385, 'serial_number': 'BLACKMAGIC', 'location': '1-5:x.0', 'manufacturer': 'Microsoft', 'product': None, 'interface': None}
scons: *** [phony_get_blackmagic] Blackmagic probe not found
```
And with your suggested fix, it fails to find the proper one.

**Review by skotopes — CHANGES_REQUESTED** [review]

**Comment by skotopes** [comment]
Un-draft when ready

File: scripts/fbt_tools/blackmagic.py (hunk 1 above)
Comment (quen0n) [reply scripts/fbt_tools/blackmagic.py]: Okay, then will need to find another way to determine the fidelity of the blackmagic port ¯\_(ツ)_/¯

Comment (quen0n) [reply scripts/fbt_tools/blackmagic.py]: Tell me, what version of Blackmagic firmware are you using? So I can test different options

**Comment by quen0n** [comment]
I reverted the changes that were related to the definition of the Blackmagic port, since there will be a dap-link in the future. The remaining change is to skip pressing C when flashing.

**Review by skotopes — APPROVED** [review]

**Review by hedger — APPROVED** [review]

**Review by skotopes — APPROVED** [review]

**Pull request merged**